servicetalk-concurrent-api/src/main/java/io/servicetalk/concurrent/api/PublisherFlatMapMerge.java

```diff
@@ -608,27 +569,32 @@ private void throwInvalidDemand(int pendingDemand) {
 
             @Override
             public void onError(final Throwable t) {
+                Throwable currPendingError = parent.pendingError;
                 if (parent.source.maxDelayedErrors == 0) {
-                    // Make sure errors aren't delivered out of order relative to onNext signals which maybe queued.
-                    try {
-                        parent.doCancel(true, false);
-                    } finally {
-                        parent.tryEmitItem(TerminalNotification.error(t), this);
+                    if (currPendingError == null && pendingErrorUpdater.compareAndSet(parent, null, t)) {
```

Comment:
Inside the above `if` branch (line 573) `currPendingError` is never used. Consider simplifying this one to
```
if (pendingErrorUpdater.compareAndSet(parent, null, t)) {
```

Reply:
if the current thread sees a non-null we can avoid an atomic operation.